sdks/java/io/mongodb/src/main/java/org/apache/beam/sdk/io/mongodb/MongoDbGridFSIO.java

```diff
@@ -446,4 +504,164 @@ public void close() throws IOException {
       }
     }
   }
+
+
+  /**
+   * Function that is called to write the data to the give GridFS OutputStream.
+   * @param <T>
+   */
+  public interface WriteFn<T> extends Serializable {
+    /**
+     * Output the object to the given OutputStream.
+     * @param output The data to output
+     * @param outStream The OutputStream
+     */
+    void write(T output, OutputStream outStream) throws IOException;
+  }
+
+  /**
+   * A {@link PTransform} to write data to MongoDB GridFS.
+   */
+  @AutoValue
+  public abstract static class Write<T> extends PTransform<PCollection<T>, PDone> {
+    @Nullable abstract String uri();
+    @Nullable abstract String database();
+    @Nullable abstract String bucket();
```

Comment:
As these attributes are also used in the `Read`, I would recommend to create a `ConnectionConfiguration` POJO (using `AutoValue`) used by the `Read` and `Write`.